cpp/src/column/column_device_view.cu

```diff
@@ -45,54 +44,75 @@ column_device_view::column_device_view( column_view source, ptrdiff_t h_ptr, ptr
                                       source.null_count(), source.offset()},
       _num_children{source.num_children()}
 {
-  if( count_descendants(source) > _num_children ) {
-    CUDF_FAIL("Columns with grand-children are not currently supported.");
-  }
-  if( _num_children > 0 )
+  size_type num_children = source.num_children();
+  if( num_children > 0 )
   {
-    column_device_view* h_column = reinterpret_cast<column_device_view*>(h_ptr);
-    column_device_view* d_column = reinterpret_cast<column_device_view*>(d_ptr);
-    d_children = d_column;
+    auto h_column = reinterpret_cast<column_device_view*>(h_ptr);
+    auto d_column = reinterpret_cast<column_device_view*>(d_ptr);
+    auto h_end = reinterpret_cast<int8_t*>(h_column + num_children);
```

Comment:
Why use a byte-sized pointer for `h_end`? Is the `extent()` method implying that some children may be a type/size other than `column_device_view`, or in other words that the total size may not be equal to `count_descendants(source) * sizeof(column_device_view)`?

Reply:
The `extent()` method was not introduced as part of this PR.
The purpose of it was to isolate the size calculation inside a member function on the class itself. Previously, I think we were doing the `count_descendants(.)*sizeof(.)` like calculation inside of `table_view` which meant `table_view` had intimate knowledge the internals of `column_device_view`. The `extent()` method isolates the logic for the size calculation of `column_device_view` inside of the `column_device_view` class itself. Now changes to this calculation (like the one for this PR) only require updating this function and not inside of `table_view`.

Reply:
If the `column_device_view` is guaranteed to be a fixed size, I don't see what the problem is with relying on `sizeof(column_device_view)` and `count_descendants()` to determine the size of the buffer. Anyway, the part that I'm referring to is inside the `column_device_view` constructor, not part of `table_view`.
My concern is more that I don't see why we need to be working with byte pointers here if we know we're just working with an array of `column_device_view`s. If `column_device_view` is guaranteed to be a fixed size, I think we should be just be working with `column_device_view*` and element rather than byte offsets.

Reply:
This is really outside the scope of this PR.
You are welcome to open a new issue if you want to get this modified.
Essentially `extent()` does the same computation (right now) but it is a member function that `table_view` or other non-`column_device_view` classes can call to get the size of the object.
Since `extent()` must exists, I see value in using it in the ctor rather than computing the size a different way.